### DISCUSSION TITLE
I keep getting a "filename already used" error when publishing to PyPI, even though it uploads succesfully

### DISCUSSION THREAD

**Discussion by Cikmo** [python-poetry/poetry#6834] [Q&A] [answered]
Everytime I run `poetry publish` (with a new, unused version), i get this error: 
```
HTTP Error 400: This filename has already been used, use a different version.
```
However, when I go check on PyPI, the new version has successfully been uploaded. I don't know if it's attempting to double upload or something? Any ideas?

It's highly annoying to keep getting warnings from github actions telling me the task failed when it doesnt.

**Comment by neersighted** [comment]
You have not provided enough details to understand or reproduce this. Could you please provide the name of the PyPI package as well as your Poetry version, at a minimum? A link to the project you are publishing/the GHA workflow files and run logs would also be quite helpful as well.

**Reply by Cikmo** [reply]
https://pypi.org/project/pnwapi/
https://github.com/Cikmo/pnwapi
Workflow: https://github.com/Cikmo/pnwapi/blob/master/.github/workflows/python-publish.yml

Using Poetry v1.2.1

Workflow logs:
```
Run poetry version $(git describe --tags --abbrev=0)
Bumping version from 0.0.0 to v0.0.3
Creating virtualenv pnwapi-86WkNGu7-py3.[10](https://github.com/Cikmo/pnwapi/actions/runs/3276540410/jobs/5392713333#step:5:11) in /home/runner/.cache/pypoetry/virtualenvs
Building pnwapi (v0.0.3)

Publishing pnwapi (v0.0.3) to PyPI
 - Uploading pnwapi-0.0.3-py3-none-any.whl 0%

 - Uploading pnwapi-0.0.3-py3-none-any.whl 100% - Uploading pnwapi-0.0.3.tar.gz 0%

 - Uploading pnwapi-0.0.3.tar.gz 100%
HTTP Error 400: This filename has already been used, use a different version. See https://pypi.org/help/#file-name-reuse for more information. | b'<html>\n <head>\n  <title>400 This filename has already been used, use a different version. See https://pypi.org/help/#file-name-reuse for more information.\n \n <body>\n  <h1>400 This filename has already been used, use a different version. See https://pypi.org/help/#file-name-reuse for more information.\n  The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>\nThis filename has already been used, use a different version. See https://pypi.org/help/#file-name-reuse for more information.\n\n\n \n'
Error: Process completed with exit code 1.
```

This didn't just happen on Github either. When i ran `poetry publish` locally, i got the same error, yet it still ended up on PyPI.

**Reply by neersighted** [reply]
It appears that Poetry is correctly reporting a error back. Your bdist (wheel) is published, but your sdist is not.

It is possible that this is fixed by a change in Poetry 1.2.2 related to sdist filename canonicalization. Please try uploading again, but with version 1.2.2 (also, @dimbleby, could you PTAL when you have time).

If that doesn't solve it, I'm stumped without digging into this myself locally (but the next possibility in my mind, barring Poetry doing something obviously incorrect that hasn't jumped out at me, is a Warehouse bug) -- it may take some time until I can do that, but if you can reproduce this with a clean slate test project outside GHA, I would suggest opening an issue.

**Reply by Cikmo** [reply]
I solved it. Turns out it's just something weird when using a version number that's less than 0.1.0 (which is the semantic versioning standard, to be fair). Though I am not sure if that's **really** the reason, because I have used version numbers like 0.0.1 before without any problems.

**Comment by Cikmo** [comment, marked as answer]
I managed to get it working. It appears to have been an issue with the version number itself. The version numbers I were using were below v0.1.0 (0.0.1, 0.0.1, etc.), which is generally the accepted standard for the "starting" versions.

Not sure if it is a Poetry thing, or PyPI thing though. I have used version numbers below 0.1.0 before outside of Poetry, without any issues.

**Reply by neersighted** [reply]
Python package versions are solely governed by [PEP 440](https://peps.python.org/pep-0440/#public-version-identifiers) and not SemVer.

If PyPI is enforcing some sort of no-version-below-0.1.0 rule, the server error they return is pretty misleading. I would suggest opening an issue on https://github.com/pypi/warehouse asking for clarification and reporting a potential bug.

**Reply by Cikmo** [reply]
I'll do that. Thanks.

**Comment by nbro** [comment]
I'm getting this problem and

1. No package has been uploaded to my Test PyPI account (at least according to the interface: https://test.pypi.org/manage/projects/)
2. My package has version 0.1.0, so it's not this problem
3. I'm using Poetry 1.5.1
4. I'm using this command: `poetry publish -vv --build --repository testpypi`
5. I'd done `poetry config repositories.testpypi https://test.pypi.org/legacy/`
6. I'd done `poetry config pypi-token.testpypi <my-testpypi-api-token>`
